File: Arrays/3442. Maximum Difference Between Even and Odd Frequency I/3442. Maximum Difference Between Even and Odd Frequency I.js
/**
 * @param {string} s
 * @return {number}
 */
var maxDifference = function (s) {
  count = {};
  for (const char of s) {
    count[char] = (count[char] || 0) + 1;
  }

  let maxOdd = 0;
  let minEven = Infinity;

  for (const v of Object.values(count)) {
    if (v % 2 === 0) minEven = Math.min(minEven, v);
    else maxOdd = Math.max(maxOdd, v);
  }

  return maxOdd - minEven;
};

// Time: O(n)
// Space: O(1)

File: Arrays/3442. Maximum Difference Between Even and Odd Frequency I/3442. Maximum Difference Between Even and Odd Frequency I.py
class Solution:
    def maxDifference(self, s: str) -> int:
        count = Counter(s)

        max_odd = 0
        min_even = float("inf")

        for v in count.values():
            if v % 2 == 0:
                min_even = min(min_even, v)
            else:
                max_odd = max(max_odd, v)

        return max_odd - min_even

# Time: O(n)
# Space: O(1)